src/Symfony/Component/Process/Pipes/AbstractPipes.php

```diff
@@ -29,6 +29,17 @@
     /** @var bool */
     private $blocked = true;
 
+    public function __construct($input)
+    {
+        if (is_resource($input)) {
+            $this->input = $input;
+        } elseif (is_string($input)) {
+            $this->inputBuffer = $input;
```

Comment:
if the string already fits in memory, there is no gain in putting it in a `php://temp` stream